site/guides/porch-installation.md

```diff
@@ -188,6 +188,11 @@ gcloud projects add-iam-policy-binding ${GCP_PROJECT_ID} \
 gcloud iam service-accounts add-iam-policy-binding porch-server@${GCP_PROJECT_ID}.iam.gserviceaccount.com \
   --role roles/iam.workloadIdentityUser \
   --member "serviceAccount:${GCP_PROJECT_ID}.svc.id.goog[porch-system/porch-server]"
+
+# We need to associate the Kubernetes Service Account (KSA)
+# with the GSA by annotating the KSA.
+kubectl annotate serviceaccount porch-server -n porch-system \
```

Comment:
Could we use kpt to add this prior to deployment?

I'd like to start using kpt to configure and deploy our components. #3365

We have a set-annotations function.
https://catalog.kpt.dev/set-annotations/v0.1/

It would also be nice to eventually have a dedicated function for this that reads from the gcloud config.
https://github.com/GoogleContainerTools/kpt/pull/2715/files

cc @yuwenma

Reply:
The whole thing should move to a kpt package.

Reply:
Yes agree a dedicated function makes sense, to get the GSA name and do the annotation, based on the gcloud config.

Reply:
Ok, fair. We can do this then.

Reply:
This does illustrate another dependency between apps being deployed and infrastructure - this time the right cloud service accounts. This is an area of significant pain when deploying apps and so has a lot of opportunity for an improved UX.

Reply:
Kind of thinking aloud, I could imagine two packages - the app package itself, and a separate package containing the cloud provider SAs and associated roles and role bindings. When deploying the app to a given cloud provider, you deploy the associated cloud provider app to your infra cluster (CC in the case of GCP). We would want to somehow capture the dependency, and run the appropriate version of the workload identity function (the annotation names and values are different for different cluster providers).

Reply:
I completely agree we need to provide kpt packages for our components. We should also look at hosting and managing our kpt packages through a porch setup.

Allowing users to use the gcloud information from the environment seems like a good solution here, although we also need a solution for users where they might need to specify a different project id. Similarly, users might want to use a different SA than the one provided. If so, it might be that editing the configuration directly is the best way to do it.

For porch setting up workload identity is something that will have to be configured on setup, while usage of this happens when someone registers a repo that can use WI as the auth method. For other auth schemes the user will provide the auth information when registring the repo.

Reply:
+1. I can see how much happy users will experience from getting rid of this painful KSA, GSA, WI config. Even more, if config connector is installed, the previous steps like GSA creation, IAM binding can be managed by kpt (in the form of CNRM resources). I think @justinsb has done some prior art about that.